rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -105,11 +119,31 @@ Recorder::get_missing_topics(const std::unordered_map<std::string, std::string>
   return missing_topics;
 }
 
+namespace
+{
+std::string serialized_offered_qos_profiles_for_topic(
+  std::shared_ptr<rosbag2_transport::Rosbag2Node> node,
+  const std::string & topic_name)
+{
+  YAML::Node offered_qos_profiles;
+  auto publishers_info = node->get_publishers_info_by_topic(topic_name);
+  for (auto info : publishers_info) {
+    offered_qos_profiles.push_back(rosbag2_transport::Rosbag2QoS(info.qos_profile()));
+  }
+  return YAML::Dump(offered_qos_profiles);
+}
+}  // unnamed namespace
+
 void Recorder::subscribe_topics(
   const std::unordered_map<std::string, std::string> & topics_and_types)
 {
   for (const auto & topic_with_type : topics_and_types) {
-    subscribe_topic({topic_with_type.first, topic_with_type.second, serialization_format_, ""});
+    subscribe_topic(
+      {
+        topic_with_type.first,
+        topic_with_type.second,
+        serialization_format_,
+        serialized_offered_qos_profiles_for_topic(node_, topic_with_type.first)});
```

Comment:
nit: should the braces be on a newline?

Reply:
I saw that, iirc they don't have to be.
My personal preference is that they should, but its only a preference.

Reply:
Do you mean the closing brace?

I wanted the opening brace to be

```
subscribe_topic({
```

but uncrustify doesn't like that